lib/zip_source_buffer.c

```diff
@@ -230,11 +227,7 @@ read_data(void *state, void *data, zip_uint64_t len, zip_source_cmd_t cmd) {
             return -1;
         }
 
-#if __STDC_WANT_SECURE_LIB__
-        memcpy_s(data, sizeof(ctx->attributes), &ctx->attributes, sizeof(ctx->attributes));
-#else
-        memcpy(data, &ctx->attributes, sizeof(ctx->attributes));
-#endif
+        _zip_memcpy(data, sizeof(ctx->attributes), &ctx->attributes, sizeof(ctx->attributes));
```

Comment:
Should this be `_zip_memcpy(data, sizeof(*data), &ctx->attributes, sizeof(ctx->attributes));`?